### PR TITLE
"ignore clearance" missions will offer even if player has no clearance

### LINKED ISSUE #1770: Would like a way to override mission location filter behavior of omitting planets that you don't have permission to land on
This has come up a few times when I'm writing missions. For example, say I set a destination filter as follows:

```
    destination
        attributes pirate
```

As long as the pirates are hostile to me (and they generally are 🙂), the game will see all pirate planets as unavailable, the destination filter will never return any valid planets, and the mission will not be offered.

I'd like a way to override this so that planets you're not allowed to land on will match the filter. You'd need to bribe your way onto the planet to complete the mission, but hey, imagine the possibilities! 🙂

### PR DESCRIPTION
**Pull request by UnorderedSigh**
**Feature:** Fixes #1770 by implementing its requested feature as `ignore clearance`

## Feature Details
A mission with `ignore clearance` will offer even if the player does not have clearance to land.

```ruby
mission "My Mission"
    ignore clearance
    destination
       government Pirate
```

## UI Screenshots
N/A

## Usage Examples
You can do missions like these:

1. Deliver to a pirate planet, but you have to bribe to land.
2. Go find a way to land on a planet that needs gaslining (or some other tech).
3. Make peace with government XYZ, so you can land on their planet and talk to them.

## Testing Done
1. Put this in data/[ignoretest.txt](https://github.com/endless-sky/endless-sky/files/10611967/ignoretest.txt)
2. Depart and land.
3. Visit job board.
4. Get the "Impossible Delivery" mission, to deliver cargo to the Korath.

### Automated Tests Added
N/A

## Performance Impact
N/A

### CODE REVIEW

File: source/Mission.cpp [hunk 1]
```diff
@@ -226,6 +226,8 @@ void Mission::Load(const DataNode &node)
 			clearance = (child.Size() == 1 ? "auto" : child.Token(1));
 			clearanceFilter.Load(child);
 		}
+		else if(child.Size() == 2 && child.Token(0) == "ignore" && child.Token(1) == "clearance")
```
Comment (RisingLeaf) [inline source/Mission.cpp:229]:
```suggestion
		else if(child.Size() == 2 && child.Token(0) == "ignore" && child.Token(1) == "clearance")
```
unless we want to expand on this with other ignores, this should be one token

**Review by RisingLeaf — APPROVED** [review]
Code wise this is fine, but as said, I think it should be one token

File: source/Mission.cpp (hunk 1 above)
Comment (UnorderedSigh) [reply source/Mission.cpp:229]: Yes, we will expand with more ignores: keystone, gaslining, etc. Those are harder to implement, while "clearance" is trivial.

Comment (RisingLeaf) [reply source/Mission.cpp:229]: well in that case, I am fine.

Although... how about allowing it to be a child node? On the other hand that makes not a lot of sense if we have only one, probably something to be implemented later.

Comment (UnorderedSigh) [reply source/Mission.cpp:229]: Yes, I expect people will want all sorts of additions one day. For now, I want to limit it to the minimum necessary for the feature, and avoid scope creep.

Comment (RisingLeaf) [reply source/Mission.cpp:229]: perfectly fine then.

**Review by Hurleveur — APPROVED** [review]
Looks good to me

I was thinking this mb could be copied to the instantiated mission but since it is not used after instantiation, there is no need.

**Comment by UnorderedSigh** [comment]
It is copied to the instantiated mission.

**Comment by UnorderedSigh** [comment]
Last tested and confirmed to work at Sun, 19 Feb 2023 04:31:14 UTC.

**Comment by UnorderedSigh** [comment]
Last tested and confirmed to work at Sat, 04 Mar 2023 18:05:09 UTC.

**Comment by UnorderedSigh** [comment]
Last tested and known to work Sat, 25 Mar 2023 22:35:22 UTC.

**Comment by UnorderedSigh** [comment]
A test is failing due to sound card driver errors again.

**Comment by UnorderedSigh** [comment]
Last tested and known to work Sun, 16 Apr 2023 23:32:02 UTC, apart from sound card errors in the Azure virtual machines. Those are unrelated to my changes.

**Comment by UnorderedSigh** [comment]
It seems the sound card issues have resolved themselves. All of the tests have passed.

**Review by petervdmeer — APPROVED** [review]

**Pull request merged**